### PR TITLE
blas/gonum: add Ztbmv

### PR DESCRIPTION
**Pull request by vladimir-ch**
Please take a look.

Updates #184 

I don't understand how it happened that Zhbmv was not gofmted and that is was not detected.

### CODE REVIEW

File: blas/gonum/level2cmplx128.go [hunk 1]
```diff
@@ -1239,6 +1239,201 @@ func (Implementation) Zhpr2(uplo blas.Uplo, n int, alpha complex128, x []complex
...
+						x[ix] += x[jx] * aij
+						jx += incX
+					}
+					ix += incX
+				}
+			}
+		} else {
+			if incX == 1 {
+				for i := n - 1; i >= 0; i-- {
+					kk := min(k, i)
```
Comment (kortschak) [inline blas/gonum/level2cmplx128.go]: Move this down.

File: blas/gonum/level2cmplx128.go [hunk 2]
```diff
@@ -1239,6 +1239,201 @@ func (Implementation) Zhpr2(uplo blas.Uplo, n int, alpha complex128, x []complex
...
+						x[i] *= ab[i*ldab+k]
+					}
+					for j, aij := range ab[i*ldab+k-kk : i*ldab+k] {
+						x[i] += x[i-kk+j] * aij
+					}
+				}
+			} else {
+				ix := kx + (n-1)*incX
+				for i := n - 1; i >= 0; i-- {
+					kk := min(k, i)
```
Comment (kortschak) [inline blas/gonum/level2cmplx128.go]: Move this down.

File: blas/gonum/level2cmplx128.go [hunk 3]
```diff
@@ -1239,6 +1239,201 @@ func (Implementation) Zhpr2(uplo blas.Uplo, n int, alpha complex128, x []complex
...
+					}
+					ix += incX
+				}
+			}
+		} else {
+			if incX == 1 {
+				for i := n - 1; i >= 0; i-- {
+					kk := min(k, i)
+					if diag == blas.NonUnit {
+						x[i] *= ab[i*ldab+k]
```
Comment (kortschak) [inline blas/gonum/level2cmplx128.go]: Is there a benefit to working on `tmp := x[i]` and restoring in these loops? This takes potentially `n` bounds checks to 2 bounds checks for these loops.

File: blas/gonum/level2cmplx128.go [hunk 4]
```diff
@@ -1239,6 +1239,201 @@ func (Implementation) Zhpr2(uplo blas.Uplo, n int, alpha complex128, x []complex
...
+						x[i] *= ab[i*ldab]
+					}
+					for j, aij := range ab[i*ldab+1 : i*ldab+kk+1] {
+						x[i] += x[i+j+1] * aij
+					}
+				}
+			} else {
+				ix := kx
+				for i := 0; i < n; i++ {
+					kk := min(k, n-i-1)
```
Comment (kortschak) [inline blas/gonum/level2cmplx128.go]: Move down.

File: blas/gonum/level2cmplx128.go [hunk 5]
```diff
@@ -1239,6 +1239,201 @@ func (Implementation) Zhpr2(uplo blas.Uplo, n int, alpha complex128, x []complex
...
+	if incX < 0 {
+		kx = (1 - n) * incX
+	}
+
+	switch trans {
+	case blas.NoTrans:
+		if uplo == blas.Upper {
+			if incX == 1 {
+				for i := 0; i < n; i++ {
+					kk := min(k, n-i-1)
```
Comment (kortschak) [inline blas/gonum/level2cmplx128.go]: Move down.

File: blas/gonum/level2cmplx128.go (hunk 1 above)
Comment (vladimir-ch) [reply blas/gonum/level2cmplx128.go]: Done at all 4 places.

File: blas/gonum/level2cmplx128.go (hunk 3 above)
Comment (vladimir-ch) [reply blas/gonum/level2cmplx128.go]: There probably is although I didn't measure it. Originally I wrote it for conciseness but also with the temporary variables it's clear enough, so I've changed the code as you suggest.

**Comment by vladimir-ch** [comment]
I have also switched the test to use subtests, only recently I realized that we can finally use them.

PTAL

**Pull request merged**